components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -416,6 +427,81 @@ protected JWTClaimsSet createJWTClaimSet(OAuthAuthzReqMessageContext authAuthzRe
         return jwtClaimsSet;
     }
 
+    /**
+     * To get authenticated subject identifier.
+     *
+     * @param authAuthzReqMessageContext Auth Request Message Context.
+     * @param tokenReqMessageContext     Token request message context.
+     * @return authenticated subject identifier.
+     */
+    private String getAuthenticatedSubjectIdentifier(OAuthAuthzReqMessageContext authAuthzReqMessageContext,
+            OAuthTokenReqMessageContext tokenReqMessageContext) {
+
+        AuthenticatedUser authenticatedUser;
+        if (authAuthzReqMessageContext != null) {
+            authenticatedUser = authAuthzReqMessageContext.getAuthorizationReqDTO().getUser();
+        } else {
+            authenticatedUser = tokenReqMessageContext.getAuthorizedUser();
+        }
+        return authenticatedUser != null ? authenticatedUser.getAuthenticatedSubjectIdentifier() : null;
```

Comment:
In the null case shouldn't we break the flow?
We need to decide whether a self-contained access token with no subject confirmation is acceptable.

Reply:
The relevant authenticated user will be set by relevant grant handlers. Hence the resposibility is purely dependant on the grant handlers. Further I moved this code to separate method from previously existing code segment.

Reply:
Throwing the exception in the event authenticated user is null